rust/libnewsboat/src/utils.rs

```diff
@@ -265,6 +265,43 @@ pub fn unescape_url(rs_str: String) -> Option<String> {
     Some(result.unwrap().replace("\0",""))
 }
 
+pub fn make_title(rs_str: String) -> String {
+    /* Sometimes it is possible to construct the title from the URL
+     * This attempts to do just that. eg:
+     * http://domain.com/story/yy/mm/dd/title-with-dashes?a=b
+     */
+    // Strip out trailing slashes
+    let mut result = rs_str.trim_right_matches('/');
+
+    // get to the final part of the URI's path and
+    // extract just the juicy part 'title-with-dashes?a=b'
+    let v: Vec<&str> = result.rsplitn(2, '/').collect();
+    result = v[0];
```

Comment:
Because `result` cannot be a null string. Since split always returns the given string the "worst case" for the vector access is "" which is fine.
I'm not sure this or the next one needs a comment since I would just be explaining how `splitn()` works.
If you insist I will gladly put it in though.